src/ethereum_test_tools/common/types.py

```diff
@@ -349,26 +349,25 @@ class KeyValueMismatch(Exception):
         was different.
         """
 
+        address: str
         key: int
         want: int
         got: int
 
-        def __init__(self, key: int, want: int, got: int, *args):
+        def __init__(self, address: str, key: int, want: int, got: int, *args):
             super().__init__(args)
+            self.address = address
             self.key = key
             self.want = want
             self.got = got
 
         def __str__(self):
             """Print exception string"""
             return (
-                "incorrect value for key {0}: want {1} (dec:{2})," + " got {3} (dec:{4})"
-            ).format(
-                Storage.key_value_to_string(self.key),
-                Storage.key_value_to_string(self.want),
-                self.want,
-                Storage.key_value_to_string(self.got),
-                self.got,
+                f"incorrect value in address {self.address} for "
```

Comment:
:clap: I was really missing this info yesterday while working on the transient storage tests!